bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,177 @@
+            and min_unique_letters < len(set(word)) < max_unique_letters
+        ]
+
+        if not filtered_words:
+            filter_not_found_embed = Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No words could be found that fit all the filters specified.",
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=filter_not_found_embed)
+            return
+
+        word = choice(filtered_words)
+
+        user_guess = "_" * len(word)
+        tries = 6
+        guessed_letters = set()
+
+        # check if the game is singleplayer or multiplayer
+        def check(msg: Message) -> bool:
+            if singleplayer == 's':
+                return msg.author == ctx.author
+            elif singleplayer == 'm':
+                return msg.author != self.bot
+            else:
+                raise commands.BadArgument("`singleplayer` must be either `s` or `m`")
+
+        hangman_embed = Embed(
+            title="Hangman",
+            color=Colours.python_blue,
+        )
+        hangman_embed.set_image(url=IMAGES[tries])
+        hangman_embed.add_field(
+            name=f"You've guessed `{user_guess}` so far.",
+            value="Guess the word by sending a message with the letter!",
+            inline=False,
+        )
+        hangman_embed.set_footer(text=f"Tries: {tries}")
+        original_message = await ctx.send(embed=hangman_embed)
+
+        while user_guess != word:
+            try:
+                message = await self.bot.wait_for(
+                    event="message",
+                    timeout=30.0,
+                    check=check
+                )
+            except TimeoutError:
+                timeout_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="Looks like the bot timed out! You must send a letter within 60 seconds.",
+                    color=Colours.soft_red,
+                )
+                await original_message.edit(embed=timeout_embed)
+                return
+
+            message.content = message.content.lower()
+            if len(message.content) > 1:
+                letter_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description="You can only send a letter for the hangman game!",
+                    color=Colours.soft_red,
+                )
+                await original_message.edit(embed=letter_embed)
+                continue
+
+            elif message.content in guessed_letters:
+                already_guessed_embed = Embed(
+                    title=choice(NEGATIVE_REPLIES),
+                    description=f"You have already guessed `{message.content}`!",
+                    color=Colours.soft_red,
+                )
+                await original_message.edit(embed=already_guessed_embed)
+                continue
+
+            elif message.content in word:
```

Comment:
+1

Reply:
It changes message.content to message.content.lower() on line 119

Reply:
Oh sorry didn't see that, though I don't think that overwriting the original message content is a good idea, as this make the code less readable...

Reply:
> Oh sorry didn't see that, though I don't think that overwriting the original message content is a good idea, as this make the code less readable...

No worries, I can see your point though each if statement would be repetitive a bit

Reply:
You could make a new variable for the normalised content

Reply:
Yea, it could be easier to read if you did `lowered_content = message.content.lower()` and then used `lowered_content` in each of the if statements, rather than editing an attr on the message itself.